apps/dot_ext/views/authorization.py

```diff
@@ -85,3 +86,19 @@ def dispatch(self, request, uuid, *args, **kwargs):
             approval.save()
 
         return result
+
+
+class TokenView(DotTokenView):
+    """
+    Override the base token view from dot to
+    handle error on refresh with better 404 error description.
+    """
+
+    def post(self, request, *args, **kwargs):
+
+        try:
+            response = super().post(request, *args, **kwargs)
+        except Http404 as error:
+            response = HttpResponse('invalid_grant', status=404)
```

Comment:
Is it OK to assume a 404 from with in the post() is going to be an "invalid_grant", or should I validate that the related refresh_token doesn't have a related token also?